### PR TITLE
Fixed RuntimeError during the deserialization

### PR DESCRIPTION
**Pull request by Enolerobotti**
I have serialized a network which was trained on GPU. When I tried to deserialize it on the CPU machine I got the following error:

Traceback (most recent call last):
  File "C:\Anaconda3\envs\python36s\lib\site-packages\IPython\core\interactiveshell.py", line 3319, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-9-c2fb86a07d04>", line 1, in <module>
    entity=joblib.load(fn)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\joblib\numpy_pickle.py", line 585, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\joblib\numpy_pickle.py", line 504, in _unpickle
    obj = unpickler.load()
  File "C:\Anaconda3\envs\python36s\lib\pickle.py", line 1050, in load
    dispatch[key[0]](self)
  File "C:\Anaconda3\envs\python36s\lib\pickle.py", line 1398, in load_reduce
    stack[-1] = func(*args)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\storage.py", line 134, in _load_from_bytes
    return torch.load(io.BytesIO(b))
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 426, in load
    return _load(f, map_location, pickle_module, **pickle_load_args)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 613, in _load
    result = unpickler.load()
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 576, in persistent_load
    deserialized_objects[root_key] = restore_location(obj, location)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 155, in default_restore_location
    result = fn(storage, location)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 131, in _cuda_deserialize
    device = validate_cuda_device(location)
  File "C:\Anaconda3\envs\python36s\lib\site-packages\torch\serialization.py", line 115, in validate_cuda_device
    raise RuntimeError('Attempting to deserialize object on a CUDA '
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.

Here I propose the solution which works for me

Fixes #{issue number}

### CODE REVIEW

**Comment by mruberry** [comment]
Hey @Enolerobotti, thanks for the PR. However, I think the current behavior is correct, and we don't want to automatically load on the CPU for people as that might be confusing. Is it so hard to follow the instructions in the current warning?

**Comment by Enolerobotti** [comment]
Hi @mruberry, thank you for reply! It is not hard for me to follow the instructions. But there are some third party tools (like joblib in my case) which do not follow them. I could PR into joblib, but I assume there are some more tools which can use torch. I proposed this solution in order to eliminate errors like this at all. I agree, that try-except not a beautiful solution. How does sound to check whether we use a CPU-only machine and automatically load model to CPU in this case only?

**Comment by mruberry** [comment]
@Enolerobotti I've been thinking about your proposal and there are pros and cons. I'm still leery of performing an operation like this without the user being aware it's happening, and there could be a variety of errors that affect loading but this would triggered in all those cases. Also, there are other behavior to consider simultaneously, like if the user specified their own map_location, which fails for some reason, but then this load silently succeeds but performs a different operation.

I just think there are too many variables to make this change in PyTorch itself.

**Comment by Enolerobotti** [comment]
Let's close the PR then
